### PR TITLE
Fix auth error after interactive login

### PR DESCRIPTION
**Pull request by lionello**
## Description

Because of the two clients `client` and `localClient`, the CLI would crap out with an auth error even after interactive login. This is because only one of the client vars got updated.

Fixing this also uncovered a testing error in the gating tests.

Regression from #882

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added appropriate tests
- [ ] I have updated the Defang CLI docs and/or README to reflect my changes, if necessary

### CODE REVIEW

File: src/cmd/cli/command/commands_test.go [hunk 1]
```diff
@@ -144,19 +161,19 @@ func TestCommandGates(t *testing.T) {
...
 			accessAllowed: false,
-			wantError:     gating.TIER_ERROR_MESSAGE + "no access to use aws provider",
+			wantError:     "current subscription tier does not allow this action: no access to use aws provider",
 		},
 		{
 			name:          "config set - aws - no access",
 			command:       []string{"config", "set", "var", "--project-name=app", "--provider=aws", "--dry-run"},
 			accessAllowed: false,
-			wantError:     gating.TIER_ERROR_MESSAGE + "no access to use aws provider",
+			wantError:     "current subscription tier does not allow this action: no access to use aws provider",
```
Comment (lionello) [inline src/cmd/cli/command/commands_test.go:170]: always use hardcoded strings in tests, so we can catch unintentional changing of those globals @nullfunc

File: src/cmd/cli/command/commands_test.go [hunk 2]
```diff
@@ -94,28 +94,45 @@ func (s *mockStsProviderAPI) AssumeRole(ctx context.Context, params *sts.AssumeR
 	return &aro, nil
 }
 
-var stsProviderApi aws.StsProviderAPI = &mockStsProviderAPI{}
```
Comment (lionello) [inline src/cmd/cli/command/commands_test.go:97]: avoid globals if not strictly necessary

File: src/cmd/cli/command/commands_test.go [hunk 3]
```diff
@@ -94,28 +94,45 @@ func (s *mockStsProviderAPI) AssumeRole(ctx context.Context, params *sts.AssumeR
...
-var ssmClient = &MockSsmClient{}
-
-func testCommand(args []string) error {
+func testCommand(args []string, cluster string) error {
+	if cluster != "" {
+		args = append(args, "--cluster", strings.TrimPrefix(cluster, "http://"))
+	}
```
Comment (jordanstephens) [inline src/cmd/cli/command/commands_test.go:100]: Does this mean that we are testing against a live fabric now? Should we tag this test as an integration test then?

**Review by jordanstephens — APPROVED** [review]

File: src/cmd/cli/command/commands_test.go (hunk 3 above)
Comment (lionello) [reply src/cmd/cli/command/commands_test.go:100]: no it runs a test server from Go's `httptest` package which is running on localhost.

**Review by nullfunc — APPROVED** [review]

**Pull request merged**